### ISSUE TITLE
Using the remote account syntax for a local account causes the profile page to never load

### ISSUE THREAD

**Issue by j-f1** [mastodon/mastodon#22329]
### Steps to reproduce the problem

Go to https://mastodon.social/@Gargon@mastodon.social

### Expected behaviour

The page should either redirect to https://mastodon.social/@Gargon or display Eugen’s profile

### Actual behaviour

The page shows a loading spinner forever

### Detailed description

I couldn’t see any dupes of this issue after a quick search, but I also don’t know the correct names for things so I may have missed it.

### Specifications

mastodon.social (currently v4.0.2)
Safari Technology Preview Release 159 (Safari 16.4, WebKit 17615.1.12.130.1)
macOS 12.6.1 (21G217)

**Comment by nightpool** [comment]
where would you ever encounter this link as a normal user? Or is this only
an issue if you're editing the URL manually?

**Comment by j-f1** [comment]
I don’t think it would be encountered as a normal user. The way I ran into the bug was that I saw a reference to `@someone@mastodon.social` and wanted to look at their profile, so I ended up copying `@someone@mastodon.social` and pasting it into the URL bar after `mastodon.social/`.
